_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_vaapi.cpp

```diff
@@ -225,8 +226,12 @@ VABufferID& VABuffersHandler::VABuffersNew(mfxU32 id, mfxU32 pool, mfxU32 num)
 
     begin = it; idBegin = idIt;
 
+    mfxStatus mfxSts;
+
     for (;it != end && *idIt == id; it++, idIt++)
-        MFX_DESTROY_VABUFFER(*it, m_vaDisplay);
+    {
+        mfxSts = CheckAndDestroyVAbuffer(m_vaDisplay, *it);
```

Comment:
Why don't you check mfxSts here? I suppose the main idea of replacing MFX_DESTROY_VABUFFER to CheckAndDestroyVAbuffer is checking returning status every time

Reply:
Have to continue destroying another buffers and somehow report errors.

Reply:
yes, we need some kind of logging. at least we should see this status in debug